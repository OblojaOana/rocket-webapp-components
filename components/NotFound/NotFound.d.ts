export interface NotFoundProps {

    /**
     * The message to be displayed when this component is rendered
     */
    title?: string;

    /**
     * Demos:
     * 
     * - https://bit.dev/totalsoft_oss/react-mui/not-found
     * 
     */
}

export default function NotFound(props: NotFoundProps): JSX.Element;