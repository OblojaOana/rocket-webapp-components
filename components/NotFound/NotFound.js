import React from "react";
import PropTypes from "prop-types";
import Card from "../Card";

const NotFound = ({ title }) => {
  return <Card>{title}</Card>;
};

NotFound.propTypes = {
  title: PropTypes.string
};

export default NotFound;
