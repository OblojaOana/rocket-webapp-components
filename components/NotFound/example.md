##### Copy this snapshot in the overview panel

```js
import React from "react";
import NotFound from "@bit/totalsoft_oss.react-mui.not-found";

export default <NotFound title="Not found"></NotFound>;
```
