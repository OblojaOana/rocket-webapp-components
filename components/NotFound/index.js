import NotFound from "../../core/NotFound";
export default NotFound;
