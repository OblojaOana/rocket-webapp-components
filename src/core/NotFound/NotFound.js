import React from 'react';
import PropTypes from 'prop-types';
import { Card } from '@material-ui/core';

const NotFound = ({ ...props }) => {
    const { title } = props;
    return (
        <Card>
            <React.Fragment>
                {title}
            </React.Fragment>
        </Card>
    )
}

NotFound.propTypes = {
    title: PropTypes.string
}

export default NotFound;