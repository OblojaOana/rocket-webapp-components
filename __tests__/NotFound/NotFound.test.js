import React from "react";
import renderer from "react-test-renderer";
import NotFound from "../../components/NotFound/NotFound";

describe("Render NotFound", () => {
  it("render correctly NotFound component", () => {
    const component = renderer.create(<NotFound />);

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
