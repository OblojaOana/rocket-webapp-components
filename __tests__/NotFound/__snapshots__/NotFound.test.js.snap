// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NotFound render correctly NotFound component 1`] = `
<div
  className="makeStyles-card-1"
/>
`;
